beacon-chain/db/kv/state.go

```diff
@@ -104,8 +105,21 @@ func (k *Store) SaveState(ctx context.Context, state *pb.BeaconState, blockRoot
 func (k *Store) DeleteState(ctx context.Context, blockRoot [32]byte) error {
 	ctx, span := trace.StartSpan(ctx, "BeaconDB.DeleteState")
 	defer span.End()
+
+	finalizedCheckpt, err := k.FinalizedCheckpoint(ctx)
```

Comment:
Can you do this in the batch? it would make it faster with batch processing / go routines